components/suggest/src/rs.rs

```diff
@@ -648,6 +648,22 @@ pub(crate) struct DownloadedGlobalConfigInner {
     pub show_less_frequently_cap: i32,
 }
 
+pub(crate) fn deserialize_f64<'de, D>(deserializer: D) -> std::result::Result<Option<f64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    String::deserialize(deserializer).map(|s| s.parse().ok())
+}
+
+pub(crate) fn deserialize_f64_or_default<'de, D>(
+    deserializer: D,
+) -> std::result::Result<f64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(deserialize_f64(deserializer)?.unwrap_or_default())
```

Comment:
Could you use `Option<f64>::deserialize(deserializer)` instead of `deserialize_f64`?  If not, maybe that function should be renamed to `deserialize_opt_f64`.  My concern is that if someone tried to use `deserialize_f64` with a regular `f64` field, it won't work (I think -- maybe I'm not understanding this right).

Reply:
That compiles (with a slight change to `Option::<f64>`) but fails at runtime with:

```
Err(Error("invalid type: string \"34.91814\", expected f64", line: 0, column: 0))
```

Instead of having two functions, I think I'll just combine everything into `deserialize_f64_or_default()`. I copied `deserialize_f64()` from [an older revision](https://github.com/mozilla/application-services/blob/04a9195927f3e4f7df5cc8eba1d1598bc599a5d4/components/suggest/src/rs.rs#L321) back when weather data was stored inline in the RS record and we needed to parse the optional score float. I thought it might be useful to bring back, but on second thought there's probably no need.